File: 169-MajorityElement/169-MajorityElement.java
// Last updated: 8/2/2025, 4:02:52 PM
class Solution {
    public int majorityElement(int[] nums) {
         return MajorityElement(nums);
    }
    public static int MajorityElement(int[] arr) {
		int e=arr[0];
		int v=1;
		for(int i=1;i<arr.length;i++) {
			if(e==arr[i]) {
				v++;
			}
			else {
				v--;
				if(v==0) {
					e=arr[i];
					v=1;
				}
			}
		}
		return e;
	}
}

File: 169-MajorityElement/169-MajorityElement.py
# Last updated: 8/2/2025, 4:02:56 PM
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        dict = {}
        count, itm = 0, ''
        for item in reversed(nums):
            dict[item] = dict.get(item, 0) + 1
            if dict[item] >= count :
                count, itm = dict[item], item
        return(itm)